src/background/tab-state.js

```diff
@@ -140,15 +143,47 @@ export default function TabState(initialState, onchange) {
   };
 
   /**
-   * Request the current annotation count for the tab's URL
+   * Request the current annotation count for the tab's URL.
+   * It debounces previous requests within the value defined by
+   * WAITMILLISECONDS
    *
    * @method
    * @param {integer} tabId The id of the tab.
    * @param {string} tabUrl The URL of the tab.
    */
   this.updateAnnotationCount = function (tabId, tabUrl) {
-    return uriInfo.getAnnotationCount(tabUrl).then(count => {
-      this.setState(tabId, { annotationCount: count });
+    const WAITMILLISECONDS = 3000; //
```

Comment:
Might I suggest for readability `WAIT_MILLISECONDS`? Or you could abbreviate to `WAIT_MS`